src/console/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+if(ENABLE_CONSOLE_COMPILATION)
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} symbolic-ref --short HEAD 
+    OUTPUT_VARIABLE git_branch
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+
+  add_custom_command(
+    OUTPUT ${CMAKE_BINARY_DIR}/nebula-console
+    COMMAND bash ${CMAKE_SOURCE_DIR}/src/console/compile-console.sh ${git_branch} ${CMAKE_BINARY_DIR}
```

Comment:
what happens if there not exists related branch in nebula-console repo?

Reply:
Compilation failed.

Reply:
Controlled by the cmake option `ENABLE_CONSOLE_COMPILATION`.